acceptance/suites/tests/https/client_may_use_external_cert_chains.rb

```diff
@@ -0,0 +1,99 @@
+  wait_for_server = "while [[ 0 -ne `curl -ksw '%{exitcode}' 'https://#{master}:7777/' | tail -1` ]]; do echo 'sleeping and waiting'; sleep 2; done"
+
+  server_script = <<EOF
+    require 'webrick'
+    require 'webrick/https'
+    require 'openssl'
+    require 'yaml'
+    require 'puppet'
+
+    cert = OpenSSL::X509::Certificate.new(File.read('#{server_cert}'))
+    key = OpenSSL::PKey.read(File.read('#{server_key}'))
+    log_file = File.open('#{tmpdir}/webrick.log', 'a+')
+    logger = WEBrick::Log.new(log_file, WEBrick::Log::DEBUG)
+    access_log = [ [log_file, WEBrick::AccessLog::COMBINED_LOG_FORMAT] ]
+    server = WEBrick::HTTPServer.new(Port: 7777, SSLEnable: true,
+                                     SSLCertificate: cert, SSLPrivateKey: key,
+                                     Logger: logger, AccessLog: access_log)
+    trap(:TERM) { server.shutdown }
+    server.mount_proc '/' do |request, response|
+      if request.body
+        report = YAML.parse(request.body).to_ruby
+        File.write('#{directory_to_serve}/' + report.host, report.status)
+        response.status = 200
+      else
+        response.status = 400
+      end
+    end
+
+    WEBrick::Daemon.start
+
+    server.start
+EOF
+
+  enable_https_report_processor_config = {
+    'master' => {
+      'reporturl' => "https://#{master}:7777/",
+      'report_include_system_store' => true,
+      'reports' => 'store,http',
+      'ssl_trust_store' => server_cert
+    }
+  }
+
+  teardown do
+    on master, kill_server
+  end
+
+
+  # Test
+  on master, generate_self_signed_cert
+  create_remote_file(master, server_rb, server_script)
+  on master, "mkdir #{directory_to_serve}"
+  on master, "chmod +rx #{tmpdir}"
+  on master, run_server
+  on master, wait_for_server
+
+  with_puppet_running_on(master, enable_https_report_processor_config) do
+    on master, 'puppet agent -t'
+  end
+
+  report_content = on(master, "cat #{directory_to_serve}/#{master}").stdout.chomp
+  assert_match /(un)?changed/, report_content
+
+
+  if 0 == on(master, 'test -f /opt/puppetlabs/puppet/ssl/puppet-cacerts', accept_all_exit_codes: true).exit_code
+    tmpdir = master.tmpdir('system_ssl_store')
+    script_location = tmpdir + '/connection_test.rb'
+
+    test_script = <<EOF
+      require 'puppet'
+      require 'puppet/server/puppet_config'
+      require 'puppet/server/http_client'
+      require 'uri'
+
+      Puppet::Server::PuppetConfig.initialize_puppet(puppet_config: {})
+      client = Puppet.runtime[:http]
+
+      response = client.get(URI('https://github.com/index.html'), options: {include_system_store: true})
+
+      puts response.code
+EOF
+
+    create_remote_file(master, script_location, test_script)
+    status = on(master, "/opt/puppetlabs/server/apps/puppetserver/bin/puppetserver ruby #{script_location}").stdout.chomp
+    assert_match /.*200$/, status
+  else
+    logger.info "Skipping system certs because /opt/puppetlabs/puppet/ssl/puppet-cacerts is missing!"
```

Comment:
Are we hoping to remove this once that bug is fixed?

Reply:
We can. Though I have no idea what the priority is for it being fixed. I can create a comment mentioning the ticket, which is what I assume you're wondering about.

Reply:
That sounds good to me.

Reply:
Added the comment at the top of the conditional